deployment/terraform/db_operations.go

```diff
@@ -91,15 +91,27 @@ func (t *Terraform) DBStatus() (string, error) {
 		return "", err
 	}
 
-	if output.DBCluster.ClusterIdentifier == "" {
-		return "", errors.New("DB cluster identifier not found")
+	// Neither Terraform DB exists, nor any External DB is used.
+	if output.DBCluster.ClusterIdentifier == "" && t.config.ExternalDBSettings.DataSource == "" {
+		return "", errors.New("DB cluster identifier not found or no external DB is used.")
 	}
 
+	// If an external non-AWS DB is used.
+	if t.config.ExternalDBSettings.DataSource != "" && t.config.ExternalDBSettings.ClusterIdentifier == "" {
+		return "available", nil
+	}
```

Comment:
Not for this PR, but: is there a way we can test the connection to the datasource here? It would need to be from one of the app nodes, but maybe it's worth doing it to make sure all is working as expected.

Reply:
At which phase do you want to test it? This DBStatus method is purely geared towards AWS instances so that we can turn them off to save money. I think we could test this in the deployment phase, but it should automatically fail if the servers won't start.